src/main/java/com/gregtechceu/gtceu/common/item/tool/behavior/MetaMachineConfigCopyBehaviour.java

```diff
@@ -0,0 +1,204 @@
+            autoOutputItem.setAllowInputFromOutputSideItems(itemData.getBoolean(INPUT_FROM_OUTPUT_SIDE));
+        }
+        if (configData.contains(FLUID_CONFIG) && machine instanceof IAutoOutputFluid autoOutputFluid) {
+            CompoundTag fluidData = configData.getCompound(FLUID_CONFIG);
+            autoOutputFluid.setOutputFacingFluids(getRelativeDirection(originalFront, machine.getFrontFacing(),
+                    intToDirection(fluidData.getInt(DIRECTION))));
+            autoOutputFluid.setAutoOutputFluids(fluidData.getBoolean(AUTO));
+            autoOutputFluid.setAllowInputFromOutputSideFluids(fluidData.getBoolean(INPUT_FROM_OUTPUT_SIDE));
+        }
+        if (configData.contains(MUFFLED) && machine instanceof IMufflableMachine mufflableMachine) {
+            mufflableMachine.setMuffled(configData.getBoolean(MUFFLED));
+        }
+        return InteractionResult.SUCCESS;
+    }
+
+    @Override
+    public void appendHoverText(ItemStack stack, @Nullable Level level, List<Component> tooltipComponents,
+                                TooltipFlag isAdvanced) {
+        tooltipComponents.add(Component.translatable("behaviour.meta.machine.config.copy.tooltip"));
+        tooltipComponents.add(Component.translatable("behaviour.meta.machine.config.paste.tooltip"));
+        if (!stack.hasTag()) return;
+        if (Screen.hasShiftDown()) {
+            tooltipComponents.add(Component.literal(""));
+            var tag = stack.getOrCreateTag();
+            if (tag.contains(CONFIG_DATA)) {
+                var data = tag.getCompound(CONFIG_DATA);
+                if (data.contains(ORIGINAL_FRONT)) {
+                    var origFront = intToDirection(data.getInt(ORIGINAL_FRONT));
+                    if (data.contains(ITEM_CONFIG)) {
+                        var itemData = data.getCompound(ITEM_CONFIG);
+                        tooltipComponents.add(Component.translatable("behaviour.setting.output.direction.tooltip",
+                                "§6Item§r",
+                                "§e" + relativeDirectionString(origFront, intToDirection(itemData.getInt(DIRECTION)))));
+                        tooltipComponents.add(Component.translatable("behaviour.setting.item_auto_output.tooltip",
+                                "§6Item§r", (itemData.getBoolean(AUTO) ? "§aenabled§r" : "§4disabled§r")));
+                        tooltipComponents.add(Component.translatable(
+                                "behaviour.setting.allow.input.from.output.tooltip", "§6Item§r",
+                                (itemData.getBoolean(INPUT_FROM_OUTPUT_SIDE) ? "§aenabled§r" : "§4disabled§r")));
+                    }
+                    if (data.contains(FLUID_CONFIG)) {
+                        var fluidData = data.getCompound(FLUID_CONFIG);
+                        tooltipComponents.add(Component.translatable("behaviour.setting.output.direction.tooltip",
+                                "§9Fluid§r", "§e" + relativeDirectionString(origFront,
```

Comment:
change relativeDirectionString() to also return a translatable

Reply:
done